spyderlib/widgets/sourcecode/syntaxhighlighters.py

```diff
@@ -190,7 +195,33 @@ def set_color_scheme(self, color_scheme):
 
     def highlightBlock(self, text):
         raise NotImplementedError
-            
+
+    def highlight_spaces(self, text, offset=0):
+        """
+        Make blank space less apparent by setting the foreground alpha.
+        This only has an effect when 'Show blank space' is turned on.
+        Derived classes could call this function at the end of
+        highlightBlock().
+        """
+        show_blanks = self.document().defaultTextOption().flags() & QTextOption.ShowTabsAndSpaces
+        if show_blanks:
+            format_leading = self.formats.get("leading", None)
+            format_trailing = self.formats.get("trailing", None)
+            match = self.BLANKPROG.search(text, offset)
+            while match:
+                start, end = match.span()
+                start = max([0, start+offset])
+                end = max([0, end+offset])
+                if end == len(text) and format_trailing is not None:

```

Comment:
Much better, maybe just add a comment on top of this line?

`# Formats points at the start of the line... leading points`